content/posts/maths/NMPC-Two-Wheel-Bicycle-Model.md

```diff
@@ -0,0 +1,186 @@
+
+<p>
+$$
+\begin{aligned}
+&\dot{\mathbf{z}}(t) = \mathbf{g}(\mathbf{z}(t),\ \mathbf{u}(t)), \\
+\text{with}\\
+&\mathbf{z}(t) = [p_x(t),\ p_y(t),\ \psi(t),\ v_x(t),\ v_y(t),\ \omega(t)]^\mathsf{T} \text{ as the state vector}, \\
+&\mathbf{u}(t) = [\tilde{d}(t),\ \delta(t)]^\mathsf{T} \text{ as the input vector}, \\
+&\mathbf{g} : \R^{n_x} \times \R^{n_u} \rightarrow \R^{n_x}, \\
+&n_x = 6 \text{ and } n_u = 2.
+\end{aligned}
+$$
+</p>
+
+Using forward Euler method, the discretised system for some small sampling time $T \in \R_{>0}$ can be written as,
+
+<p>
+$$
+\mathbf{z}(k + 1) = \mathbf{z}(k) + T \cdot \mathbf{g}(\mathbf{z}(k), \mathbf{u}(k)), \ k \in \Z_+.
+$$
+</p>
+
+Throughout this article, the vehicle parameters are taken as described in Table 1 in [1]. They are given (estimated) as,
+
+| Param    | Value            | Param    | Value              | Param    | value                             |
+| -------- | ---------------- | -------- | ------------------ | -------- | --------------------------------- |
+| $l_f$    | $0.178$ m        | $l_r$    | $0.147$ m          | $m$      | $5.6292$ kg                       |
+| $J_z$    | $0.204$ kg m$^2$ | $B_f$    | $9.242$            | $B_r$    | $17.716$                          |
+| $C_f$    | $0.085$          | $C_r$    | $0.133$            | $D_f$    | $134.585$ N                       |
+| $D_r$    | $159.919$ N      | $C_{m1}$ | $20$ N             | $C_{m2}$ | $6.92 \times 10^{-7}$ kg s$^{-1}$ |
+| $C_{m3}$ | $3.99$ N         | $C_{m4}$ | $0.67$ kg m$^{-1}$ | $M$      | $1674$                            |
+
+Notice that the value of $C_{m2}$ is almost negligible, this will be bought up in a later section. The goal is to formulate an NMPC that minimises the distance between the desired final position $\mathbf{p}_d = [p_x^d,\ p_y^d]^\mathsf{T}$ and the position of the vehicle $\mathbf{p}_N = [p_x^N,\ p_y^N]^\mathsf{T}$ after $N$ (prediction horizon) time steps. This can also be interpreted as asking the MPC controller to take the system as close as possible to the desired final position with in a given time of $N \cdot T$ seconds. Whilst doing so, the controller shall ensure that the system obeys the state and input (safety) constraints, given as
+
+<p>
+$$
+\begin{aligned}
+&v_x \in \bm{\gamma} = [0,\ 5], \\
+&\begin{bmatrix} \ \tilde{d} \ \\ \ \delta \ \end{bmatrix}  \in \bm{\mu} = [0,\ 1] \times \left[-\frac{\pi}{3},\ \frac{\pi}{3}\right]. \\
+%&\delta \in \left[-\frac{\pi}{6},\ \frac{\pi}{6}\right].
+\end{aligned}
+$$
+</p>
+
+It should also be of priority that the system is not subjected to violently arbitrary accelerations and steering angles. This can be formulated as aiming for a minimum *jerk* trajectory and can be modeled as minimising the *distance* between successive input vectors. With this, the NMPC formulation takes the following form,
+
+<p>
+$$
+\begin{aligned}
```

Comment:
I would:

1. Drop the index $i$
2. Put $u_{\rm previous}$ in the list of parameters, i.e., $P_N(z, u_{\rm previous})$